### PR TITLE
Adjust logic when finding approx matches for model or test matching

### PR DESCRIPTION
**Pull request by emilieschario**
resolves # https://github.com/dbt-labs/dbt-core/issues/3835

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

**Comment by emilieschario** [comment]
@jtcohen6 the failing flake 8 test seems unrelated (it's whitespace related on a file I'm not editing?) Is it okay to ignore?

File: core/dbt/adapters/base/relation.py [hunk 1]
```diff
@@ -89,7 +89,10 @@ def matches(
             if not self._is_exactish_match(k, v):
                 exact_match = False
 
-            if self.path.get_lowered_part(k) != v.lower():
+            if (
+                self.path.get_lowered_part(k).strip(self.quote_character) !=
```
Comment (jtcohen6) [inline core/dbt/adapters/base/relation.py:93]: I believe the flake8 error says:
```
core/dbt/adapters/base/relation.py:93:77: W291 trailing whitespace
```
which just looks like:
```suggestion
                self.path.get_lowered_part(k).strip(self.quote_character) !=
```

**Comment by emilieschario** [comment]
I completely misread the error 🤦‍♀️ . thanks @jtcohen6

**Review by jtcohen6 — APPROVED** [review]
Thanks @emilieschario :)

**Pull request merged**